src/hooks/useTokenMetadata.ts

```diff
@@ -0,0 +1,18 @@
+import { find } from 'lodash';
+import { Types } from 'rainbow-token-list';
+import { useMemo } from 'react';
+import { toChecksumAddress } from '../handlers/web3';
+import { RAINBOW_TOKEN_LIST } from '@rainbow-me/references';
+
+export default function useTokenMetadata(
+  tokenAddress: string
+): Types.Token | undefined {
+  return useMemo(
+    () =>
+      find(RAINBOW_TOKEN_LIST.tokens, [
```

Comment:
Given that we are scanning thru RAINBOW_TOKEN_LIST tokens for each time we would like to get the token metadata, we should add a const in references/uniswap/index that just does the keyBy once and holds this interim mapping of address -> metadata (so this function can be O(1) and not O(n))